pkg/actuators/machine/utils.go

```diff
@@ -16,39 +22,176 @@ const (
+)
+
+// Add the category for installer clueanup the Machine VM at cluster torn-down time
+// if the category exists in PC.
+func addCategory(mscp *machineScope, vmMetadata *nutanixClientV3.Metadata) error {
+	var err error
+	clusterID, ok := getClusterID(mscp.machine)
+	if !ok || clusterID == "" {
+		err = fmt.Errorf("%s: failed to get the clusterID", mscp.machine.Name)
+		klog.Error(err.Error())
+		return err
+	}
+
+	categoryKey := fmt.Sprintf("%s%s", NutanixCategoryKeyPrefix, clusterID)
+	_, err = mscp.nutanixClient.V3.GetCategoryValue(categoryKey, NutanixCategoryValue)
+	if err != nil {
+		klog.Errorf("%s: failed to find the category with key %q and value %q. %w", mscp.machine.Name, categoryKey, NutanixCategoryValue, err)
+		return err
+	}
+
+	// Add the category to the vm metadata
+	if vmMetadata.Categories == nil {
+		vmMetadata.Categories = make(map[string]string, 1)
+	}
+	vmMetadata.Categories[categoryKey] = NutanixCategoryValue
+
+	return nil
+}
+
+// Condition types for an Nutanix VM instance.
+const (
+	// MachineCreation indicates whether the machine's VM has been created or not. If not,
+	// it should include a reason and message for the failure.
+	MachineCreation string = "MachineCreation"
+
+	// MachineUpdate indicates whether the machine's VM has been updated or not. If not,
+	// it should include a reason and message for the failure.
+	MachineUpdate string = "MachineUpdate"
+
+	// MachineDeletion indicates whether the machine's VM has been deleted or not. If not,
+	// it should include a reason and message for the failure.
+	MachineDeletion string = "MachineDeletion"
+
+	// MachineInstanceReady indicates whether the machine's VM gets ready.
+	MachineInstanceReady string = "MachineInstanceReady"
+)
+
+// Condition reasons for the condition's last transition.
+const (
+	// MachineCreationSucceeded indicates machine creation success.
+	MachineCreationSucceeded string = "MachineCreationSucceeded"
+	// MachineCreationFailed indicates machine creation failure.
+	MachineCreationFailed string = "MachineCreationFailed"
+	// MachineUpdateSucceeded indicates machine update success.
+	MachineUpdateSucceeded string = "MachineUpdateSucceeded"
+	// MachineUpdateFailed indicates machine update failure.
+	MachineUpdateFailed string = "MachineUpdateFailed"
+	// MachineDeletionSucceeded indicates machine deletion success.
+	MachineDeletionSucceeded string = "MachineDeletionSucceeded"
+	// MachineDeletionFailed indicates machine deletion failure.
+	MachineDeletionFailed string = "MachineDeletionFailed"
+	// MachineInstanceIsReady indicates the machine's VM gets ready
+	MachineInstanceIsReady string = "Machine instance is ready"
+	// MachineInstanceNotReady indicates the machine's VM is not ready
+	MachineInstanceNotReady string = "Machine instance is not ready"
+	// MachineInstanceReadyUnknown indicates unknown if the machine's VM is ready
+	MachineInstanceReadyUnknown string = "Unknown machine instance ready or not"
+)
```

Comment:
Do these need to be public or could they be private?

Reply:
Changed to private, since all the references of these are in the same package.